src/main/java/com/rogerr/custom/controller/UserController.java

```diff
@@ -0,0 +1,120 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.rogerr.custom.model.User;
+import com.rogerr.custom.service.UserService;
+
+
+@RestController
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    private static Logger logger = LoggerFactory.getLogger(UserController.class);
+
+    // -------------------Retrieve All Users--
+
+    @RequestMapping(value = "/user/", method = RequestMethod.GET)
+    public ResponseEntity<?> listAllUsers() {
+        List<User> users = userService.findAllUsers();
+        if (users.isEmpty()) {
+            return new ResponseEntity<List<User>>(HttpStatus.NO_CONTENT);
+        }
+        return new ResponseEntity<List<User>>(users, HttpStatus.OK);
+    }
+
+    // -------------------Retrieve Single User--
+
+    @RequestMapping(value = "/user/{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getUser(@PathVariable("id") long id) {
+        logger.debug("Fetching User with id " + id);
+        User user = userService.findById(id);
+        if (user == null) {
+            logger.debug("User with id " + id + " not found");
+            return new ResponseEntity<User>(HttpStatus.NO_CONTENT);
+        }
+        return new ResponseEntity<User>(user, HttpStatus.OK);
+    }
+
+    // -------------------Create a User--
+
+    @RequestMapping(value = "/user/", method = RequestMethod.POST)
+    public ResponseEntity<?> createUser(@RequestBody User user, UriComponentsBuilder ucBuilder) {
+        logger.debug("Creating User " + user.getUsername());
+
+        if (userService.isUserExist(user)) {
+            logger.debug("A User with name " + user.getUsername() + " already exist");
+            return new ResponseEntity<Void>(HttpStatus.CONFLICT);
+        }
+
+        userService.saveUser(user);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setLocation(ucBuilder.path("/user/{id}").buildAndExpand(user.getId()).toUri());
+        return new ResponseEntity<Void>(headers, HttpStatus.CREATED);
+    }
+
+    // ------------------- Update a User--
+
+    @RequestMapping(value = "/user/{id}", method = RequestMethod.PUT)
+    public ResponseEntity<?> updateUser(@PathVariable("id") long id, @RequestBody User user) {
+        logger.debug("Updating User " + id);
+
+        User currentUser = userService.findById(id);
+
+        if (currentUser == null) {
+            logger.debug("User with id " + id + " not found");
+            return new ResponseEntity<User>(HttpStatus.NO_CONTENT);
+        }
+
+        currentUser.setUsername(user.getUsername());
+        currentUser.setAddress(user.getAddress());
+        currentUser.setEmail(user.getEmail());
```

Comment:
I can understand this might make sense if you haven't added persistence yet, however, after task 2, your objects should have a reference to their own id (if they have been persisted before), so you can just pass the whole object (with it's updates) to the service and most repository implementations are smart enough to handle the update for you.